roaringbitmap/src/main/java/org/roaringbitmap/WordStorage.java

```diff
@@ -0,0 +1,11 @@
+package org.roaringbitmap;
+
+public interface WordStorage<T> {
+
+  T add(short value);
+
+  boolean isEmpty();
+
+  T runOptimize();
```

Comment:
I know it's little too late but I've noticed this.
I think this method should be called more general `toEfficientContainer` and have that logic implemented.
Right now if we "optimiseForBitmaps", we may end up with BitmapContainers with cardinality lower than 4096 in the final bitmap which should not happen.
Also it's happy coincidence that `RunContainer::runOptimize` calls `RunContainer::toEfficientContainer`

Reply:
Yes, that was my second question why there is both `constantMemory` and `optimiseForBitmaps`. I'd remove `optimiseForBitmaps` if we still can.